packages/context/src/context.ts

```diff
@@ -116,18 +117,32 @@ export class Context extends EventEmitter {
    * upon `bind` and `unbind` events
    */
   private setupEventHandlers() {
+    // tslint:disable-next-line:no-any
+    const notificationErrorHandler = (...args: any[]) => {
+      // Catch error to avoid lint violations
+      this._debug(...args);
+      // Bubbling up the error event over the context chain
+      // until we find an error listener
+      let ctx: Context | undefined = this;
+      while (ctx) {
+        if (ctx.listenerCount('error')) {
+          this._debug('Emitting error to context %s', ctx.name);
+          ctx.emit('error', ...args);
+          break;
```

Comment:
Have you considered using early `return` here? Then you don't have have to check `if (ctx === undefined)` on L135 below.